core/src/test/java/ai/timefold/solver/core/impl/domain/solution/cloner/gizmo/GizmoCloningUtilsTest.java

```diff
@@ -0,0 +1,34 @@
+package ai.timefold.solver.core.impl.domain.solution.cloner.gizmo;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.Map;
+
+import ai.timefold.solver.core.impl.testdata.domain.clone.deepcloning.AnnotatedTestdataVariousTypes;
+import ai.timefold.solver.core.impl.testdata.domain.clone.deepcloning.ExtraDeepClonedObject;
+import ai.timefold.solver.core.impl.testdata.domain.clone.deepcloning.TestdataDeepCloningEntity;
+import ai.timefold.solver.core.impl.testdata.domain.clone.deepcloning.TestdataVariousTypes;
+import ai.timefold.solver.core.impl.testdata.domain.clone.deepcloning.field.TestdataFieldAnnotatedDeepCloningEntity;
+import ai.timefold.solver.core.impl.testdata.domain.clone.deepcloning.field.TestdataFieldAnnotatedDeepCloningSolution;
+
+import org.junit.jupiter.api.Test;
+
+class GizmoCloningUtilsTest {
```

Comment:
If my memory serves, the cloning tests are usually written in a way that the same bug can be tested for both reflection-based and Gizmo-based cloners. It is important that we maintain functional equivalence between the two implementations.

Please find the cloner test which implements this duality, and preferrably add the test there.

Reply:
Found the existing test (which fails without the change to GizmoCloningUtils), but I'll keep this one; the error message for this one failing (missing class in set) is more understandable than the one for the shared test (missing method in generated class).